channel-interceptor/src/test/scala/fr/acinq/eclair/plugins/channelinterceptor/OpenChannelInterceptorSpec.scala

```diff
@@ -0,0 +1,87 @@
+import akka.actor.typed.scaladsl.adapter.ClassicActorRefOps
+import com.typesafe.config.ConfigFactory
+import fr.acinq.bitcoin.scalacompat.Crypto.{PrivateKey, PublicKey}
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Crypto, Satoshi, SatoshiLong}
+import fr.acinq.eclair.blockchain.fee.FeeratePerKw
+import fr.acinq.eclair.channel.ChannelFlags
+import fr.acinq.eclair.io.OpenChannelInterceptor.{DefaultParams, OpenChannelNonInitiator}
+import fr.acinq.eclair.io.Peer
+import fr.acinq.eclair.io.Peer.ChannelId
+import fr.acinq.eclair.router.Router.{GetNode, PublicNode, UnknownNode}
+import fr.acinq.eclair.wire.protocol._
+import fr.acinq.eclair.{AcceptOpenChannel, CltvExpiryDelta, Features, InterceptOpenChannelCommand, InterceptOpenChannelReceived, InterceptOpenChannelResponse, MilliSatoshiLong, RejectOpenChannel, TestConstants, TimestampSecondLong, UInt64, randomBytes32, randomBytes64}
+import org.scalatest.Outcome
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+import scodec.bits.ByteVector
+
+class OpenChannelInterceptorSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("application")) with FixtureAnyFunSuiteLike {
+  val publicKey: Crypto.PublicKey = PrivateKey(ByteVector32.One).publicKey
+  val fundingAmount: Satoshi = 1 sat
+  val remoteNodeId: Crypto.PublicKey = PrivateKey(ByteVector32.One).publicKey
+  val temporaryChannelId: ByteVector32 = ByteVector32.Zeroes
+  val openChannel: OpenChannel = OpenChannel(ByteVector32.Zeroes, temporaryChannelId, fundingAmount, 0 msat, 1 sat, UInt64(1), 1 sat, 1 msat, FeeratePerKw(1 sat), CltvExpiryDelta(1), 1, publicKey, publicKey, publicKey, publicKey, publicKey, publicKey, ChannelFlags.Private)
+  val openDualChannel: OpenDualFundedChannel = OpenDualFundedChannel(ByteVector32.Zeroes, ByteVector32.One, FeeratePerKw(5000 sat), FeeratePerKw(4000 sat), 250_000 sat, 500 sat, UInt64(50_000), 15 msat, CltvExpiryDelta(144), 483, 650_000, publicKey(1), publicKey(2), publicKey(3), publicKey(4), publicKey(5), publicKey(6), publicKey(7), ChannelFlags(true))
+  val defaultParams: DefaultParams = DefaultParams(100 sat, 100000 msat, 100 msat, CltvExpiryDelta(288), 10)
+  val channels: Map[Peer.FinalChannelId, actor.ActorRef] = Map(Peer.FinalChannelId(randomBytes32()) -> system.classicSystem.deadLetters)
+  val connectedData: Peer.ConnectedData = Peer.ConnectedData(NodeAddress.fromParts("1.2.3.4", 42000).get, system.classicSystem.deadLetters, Init(TestConstants.Alice.nodeParams.features.initFeatures()), Init(TestConstants.Bob.nodeParams.features.initFeatures()), channels.map { case (k: ChannelId, v) => (k, v) })
+  val openChannelNonInitiator: OpenChannelNonInitiator = OpenChannelNonInitiator(remoteNodeId, Left(openChannel), connectedData.localFeatures, connectedData.remoteFeatures, connectedData.peerConnection.toTyped)
+  val bobAnnouncement: NodeAnnouncement = announcement(TestConstants.Bob.nodeParams.nodeId)
+
+  def publicKey(fill: Byte): PublicKey = PrivateKey(ByteVector.fill(32)(fill)).publicKey
```

Comment:
Like what we did in the eclair PR, this can be greatly simplified, some of these values are unused or used only once, it deserves a clean-up to be less confusing and focus only on the test itself.

Reply:
Indeed, cleaned up in [dc34b5a](https://github.com/ACINQ/eclair-plugins/pull/4/commits/dc34b5a581a2db7ad24bd3d7c0f2f36442917bc7).